### PR TITLE
test: continuing offers in orchestrate async-flows

### LINKED ISSUE #9673: `orchestrate` (async-flow) does not support the standard `ContinuingOfferResult`
## What is the Problem Being Solved?

`orchestrate` (async-flow) does not currently support the standard `ContinuingOfferResult`:
https://github.com/Agoric/agoric-sdk/blob/3af2b5c8660d1fb7af217183bffc2f8de0e1cbc5/packages/smart-wallet/src/types.d.ts#L32-L35

More specifically, `TopicRecordShape` (and `PublicTopicShape`) return a promise for `storagePath`:
https://github.com/Agoric/agoric-sdk/blob/c51ea3de22f50e05fcc1aaabd2108e785d51eb2e/packages/zoe/src/contractSupport/topics.js#L7-L12
https://github.com/Agoric/agoric-sdk/blob/940d3f0a993ca45a6bb0893bd59e6df1f22d9143/packages/inter-protocol/src/vaultFactory/vaultHolder.js#L91
https://github.com/Agoric/agoric-sdk/blob/940d3f0a993ca45a6bb0893bd59e6df1f22d9143/packages/zoe/src/contractSupport/recorder.js#L83-L93

### PR DESCRIPTION
**Pull request by 0xpatrickdev**
closes: #9673
refs: #9066
refs: #9042 

## Description
- Adds `basic-flow.contract.js`, proposal, and builder for testing `orchestrate` (`async-flow`) continuing invitations to facilitate testing. These are also the first bootstrap + multichain tests with `orchestrate` / `async-flow`.
 - Updates `asContinuingInvitation` and `getPublicTopics` methods to return Vows to satisfy membrane constraints
    - Ensures `storagePath`, part of `ContinuingOfferResult` is a string instead of a Promise/Vow so `smart-wallet` can easily handle this and the `async-flow` membrane is appeased (no promises)
- Updates chain facade implementations, ensuring each account has it's own storage node (towards #9066)

### Testing Considerations
Includes unit, bootstrap, and multichain (e2e) tests for an async-flow contract. E2E ensure accounts can be created on agoric, cosmos, and osmosis and return continuing invitations.

### Upgrade Considerations
This did not require any changes to smart-wallet. There is a change to `PublicTopicShape` to accept a promise or a string (previously just a promise), but this is only in contracts relaying on smart-wallet, not smart-wallet itself.

### CODE REVIEW

File: packages/orchestration/src/examples/kitchen-sink.contract.js
Comment (0xpatrickdev) [inline packages/orchestration/src/examples/kitchen-sink.contract.js]: Feedback welcome if anyone can think of a better name than `kitchen-sink`

File: multichain-testing/yarn.lock
Comment (0xpatrickdev) [inline multichain-testing/yarn.lock:1941]: How / when can we make the Multichain E2E Testing job part of `force-integration`?

Comment (LuqiPan) [reply multichain-testing/yarn.lock:1941]: @toliaqat, happen to know how we can add Multichain E2E Testing job to `force-integration`?

Based on [history](https://github.com/Agoric/agoric-sdk/actions/workflows/multichain-e2e.yml), successful runs are all under 17 mins and failed runs take several minutes.

File: packages/orchestration/src/examples/kitchen-sink.contract.js
Comment (turadg) [reply packages/orchestration/src/examples/kitchen-sink.contract.js]: A kitchen sink will be some effort to maintain. Whenever we add some functionality we'll test it in a targeted fashion and have to also include it here.

If the scope is to test async flow, maybe "stress-flows" or "basic-flows"

File: packages/orchestration/src/exos/local-chain-facade.js
Comment (turadg) [inline packages/orchestration/src/exos/local-chain-facade.js]: TODO goes at the front, and is meant to be done. If you're not sure the change should be made, make the TODO to consider it. I.e. `TODO consider`. Also in general TODO and other ALLCAPS directives go at the start

File: packages/boot/test/bootstrapTests/orchestration.test.ts
Comment (turadg) [inline packages/boot/test/bootstrapTests/orchestration.test.ts]: thank you

File: multichain-testing/test/kitchen-sink.test.ts
Comment (turadg) [inline multichain-testing/test/kitchen-sink.test.ts]: neat

File: multichain-testing/test/kitchen-sink.test.ts
Comment (turadg) [inline multichain-testing/test/kitchen-sink.test.ts]: why repeat for two users?

File: multichain-testing/test/kitchen-sink.test.ts
Comment (turadg) [inline multichain-testing/test/kitchen-sink.test.ts]: I'm curious why this is so parameterized. The macro could just take `chainName` and figure the rest of this out itself.

File: multichain-testing/tools/agd-tools.ts
Comment (turadg) [inline multichain-testing/tools/agd-tools.ts:11]: how was this discovered? do we need more CI for multichain-testing?

maybe that's why you're proposing it be in force-integration

File: packages/builders/scripts/orchestration/init-kitchen-sink.js
Comment (turadg) [inline packages/builders/scripts/orchestration/init-kitchen-sink.js]: I've been trying to reduce the cardinality of names for a core-eval script.

consider here re-using `startKitchenSink.name` by importing the starter function.

File: packages/orchestration/src/examples/kitchen-sink.contract.js
Comment (turadg) [inline packages/orchestration/src/examples/kitchen-sink.contract.js]: why not? you did add it to `orchestrationAccountMethods`. Those lists must be the same: https://github.com/Agoric/agoric-sdk/blob/21742bd3d1f6c1ec6693b2378c6eb9efa7ee9999/packages/orchestration/src/utils/orchestrationAccount.js#L11-L22

File: packages/orchestration/src/exos/cosmos-orchestration-account.js
Comment (turadg) [inline packages/orchestration/src/exos/cosmos-orchestration-account.js]: s/StoragePath/PublicTopics

File: packages/orchestration/src/exos/cosmos-orchestration-account.js
Comment (turadg) [inline packages/orchestration/src/exos/cosmos-orchestration-account.js]: please include justification

File: packages/orchestration/src/exos/local-orchestration-account.js
Comment (turadg) [inline packages/orchestration/src/exos/local-orchestration-account.js]: consider including a reason

File: packages/orchestration/src/exos/local-orchestration-account.js
Comment (turadg) [inline packages/orchestration/src/exos/local-orchestration-account.js]: s/StoragePath/PublicTopics

File: multichain-testing/tools/agd-tools.ts
Comment (0xpatrickdev) [reply multichain-testing/tools/agd-tools.ts:11]: Typescript error (warning?) in the test runner.  We don't have `lint:types` included in `lint` since it looks like:

```
Found 383 errors in 25 files.

Errors  Files
     2  ../packages/deploy-script-support/src/coreProposalBehavior.js:44
     1  ../packages/deploy-script-support/src/writeCoreEvalParts.js:74
     1  ../packages/governance/src/quorumCounter.js:17
     2  ../packages/inter-protocol/src/provisionPool.js:50
     3  ../packages/inter-protocol/src/provisionPoolKit.js:69
    12  ../packages/internal/src/callback.js:45
     1  ../packages/internal/src/lib-chainStorage.js:304
     4  ../packages/internal/src/storage-test-utils.js:122
     4  ../packages/network/src/bytes.js:10
    85  ../packages/network/src/network.js:69
    15  ../packages/network/src/router.js:53
    73  ../packages/network/src/types.js:29
     2  ../packages/swing-store/src/snapStore.js:52
     4  ../packages/SwingSet/src/kernel/state/storageHelper.js:9
    70  ../packages/vats/src/ibc.js:37
    23  ../packages/vats/src/localchain.js:34
     2  ../packages/vats/src/transfer.js:40
    20  ../packages/vow/src/E.js:61
     7  ../packages/vow/src/tools.js:17
     3  ../packages/vow/src/types.js:65
    11  ../packages/vow/src/vow-utils.js:23
     6  ../packages/vow/src/vow.js:25
     9  ../packages/vow/src/watch-utils.js:38
    18  ../packages/vow/src/watch.js:13
     5  ../packages/vow/src/when.js:7
```

Aside - a UX consideration to keep in mind for orch devs working outside agoric-sdk. Aside, aside - appreciate the work you've personally done to make these warning virtually non-existent for the bulk of `agoric-sdk/packages/*`

Comment (0xpatrickdev) [reply multichain-testing/tools/agd-tools.ts:11]: > maybe that's why you're proposing it be in force-integration

Yes, please chime in on [this thread](https://github.com/Agoric/agoric-sdk/pull/9679/files#r1672598345) to affirm or suggest an alternate plan

File: packages/orchestration/src/examples/kitchen-sink.contract.js
Comment (0xpatrickdev) [reply packages/orchestration/src/examples/kitchen-sink.contract.js]: Thanks, I will go with `basic-flows`. Will make this change in the final rebase after ✅ if that's ok

File: multichain-testing/test/kitchen-sink.test.ts
Comment (0xpatrickdev) [reply multichain-testing/test/kitchen-sink.test.ts]: No need, removed.

File: packages/builders/scripts/orchestration/init-kitchen-sink.js
Comment (0xpatrickdev) [reply packages/builders/scripts/orchestration/init-kitchen-sink.js]: This is a worthy goal. Hopefully, these can be codegen'ed in the not too far in the future.

Tangential - should we consider publishing this builder from `@agoric/orchestration` so it's more visible to devs?

File: packages/orchestration/src/exos/local-chain-facade.js
Comment (0xpatrickdev) [reply packages/orchestration/src/exos/local-chain-facade.js]: noted, done

File: packages/orchestration/src/exos/local-orchestration-account.js
Comment (0xpatrickdev) [reply packages/orchestration/src/exos/local-orchestration-account.js]: removed this comment, and will save it for the ticket instead

File: packages/orchestration/src/exos/cosmos-orchestration-account.js
Comment (0xpatrickdev) [reply packages/orchestration/src/exos/cosmos-orchestration-account.js]: Perhaps too naively, we don't care if this crosses a turn boundary. We don't expect the answer to storagePath to change.

I was a little intimidated by the phrasing in the eslint config and have reverted:
https://github.com/Agoric/agoric-sdk/blob/df05b073ad5e24c0ae794332cc602ea11afad057/.eslintrc.cjs#L111-L112

File: packages/orchestration/src/examples/kitchen-sink.contract.js
Comment (0xpatrickdev) [reply packages/orchestration/src/examples/kitchen-sink.contract.js]: I meant to come back to this. Addressed in the batch of fixups via `orchestration-api.ts`. 

Feedback on the JS Doc comments is appreciated. Our docs are a bit light on these subjects but this seemed like a good place to over-explain a little.

File: multichain-testing/test/kitchen-sink.test.ts
Comment (0xpatrickdev) [reply multichain-testing/test/kitchen-sink.test.ts]: Recognize some room for improvement here, maybe `chain`, `chainId`, `denom`, `expectedAddressPrefix` should come from support.ts function. But seems mostly harmless imo.

Looking forward to others contributing to this directory, and making improvements along the way.

File: packages/builders/scripts/orchestration/init-kitchen-sink.js
Comment (turadg) [reply packages/builders/scripts/orchestration/init-kitchen-sink.js]: > Hopefully, these can be codegen'ed

fwiw I think some helper functions will suffice

> should we consider publishing this builder

This open source repo is fully visible, and the docs site can point to it. Putting it in the NPM package creates an API to support

File: multichain-testing/test/kitchen-sink.test.ts
Comment (turadg) [inline multichain-testing/test/kitchen-sink.test.ts]: consider using the `test.macro`'s title maker and just passing the literals needed to lookup the rest. Then you don't need a loop. e.g.
```
test.serial(makeAccountScenario, 'agoric');
test.serial(makeAccountScenario, 'cosmoshub');
test.serial(makeAccountScenario, 'osmosis');
```

File: packages/orchestration/src/orchestration-api.ts
Comment (turadg) [inline packages/orchestration/src/orchestration-api.ts:186]: these are top level API changes that should have @dtribble approval. I approve and will leave to you how/when to sync with Dean

File: packages/zoe/src/contractSupport/topics.js
Comment (turadg) [inline packages/zoe/src/contractSupport/topics.js]: remove. `ERef<T>` is already `PromiseLike<T> | T` https://endojs.github.io/endo/types/_endo_eventual_send.ERef.html

**Review by turadg — APPROVED** [review]

File: multichain-testing/test/kitchen-sink.test.ts
Comment (0xpatrickdev) [reply multichain-testing/test/kitchen-sink.test.ts]: Thanks for the push to refactor the test. It led to simplifying the contract!

File: multichain-testing/yarn.lock
Comment (0xpatrickdev) [reply multichain-testing/yarn.lock:1941]: Thanks @LuqiPan @toliaqat. I am sending this PR to the merge queue, but interested to see where we land on this.

File: packages/zoe/src/contractSupport/topics.js
Comment (mhofman) [inline packages/zoe/src/contractSupport/topics.js:10]: As we discussed, changing this shape is safe against upgrade as it couldn't have been used as a durable state of collection shape before because durably storing promises is not possible.

**Pull request merged**
